packages/web/src/scss/components/FileUploader/README.md

```diff
@@ -302,6 +309,128 @@ up the template and apply it on any attachments the user wants to upload.
 </div>
 ```
 
+### Preview Image
+
+You can add a preview image to the FileUploaderAttachment.
+
+```html
+<span class="FileUploaderAttachment__image">
+  <img src="http://placekitten.com/200/300" alt="" />
+</span>
+```
+
+Full example:
+
+```html
+<li class="FileUploaderAttachment">
+  <!-- Preview image: start -->
+  <span class="FileUploaderAttachment__image">
+    <img src="http://placekitten.com/200/300" alt="" />
+  </span>
+  <!-- Preview image: end -->
+  <span class="FileUploaderAttachment__name">
+    <span class="text-truncate">My resume.docx</span>
+  </span>
+  <button type="button" class="FileUploaderAttachment__action">
+    <span class="accessibility-hidden">Remove</span>
+    <svg width="24" height="24" aria-hidden="true">
+      <use xlink:href="/icons/svg/sprite.svg#close" />
+    </svg>
+  </button>
+</li>
+```
+
+### Custom Actions
+
+You can add custom actions to the FileUploaderAttachment.
+
+```html
+<button type="button" class="FileUploaderAttachment__action">
+  <span class="accessibility-hidden">Edit</span>
+  <svg width="24" height="24" aria-hidden="true">
+    <use xlink:href="/icons/svg/sprite.svg#pencil" />
+  </svg>
+</button>
+```
+
+Full example:
+
+```html
+<li class="FileUploaderAttachment">
+  <svg width="24" height="24" aria-hidden="true">
+    <use xlink:href="/icons/svg/sprite.svg#file" />
+  </svg>
+  <span class="FileUploaderAttachment__name">
+    <span class="text-truncate">My resume.docx</span>
+  </span>
+  <!-- Custom action: start -->
+  <button type="button" class="FileUploaderAttachment__action">
+    <span class="accessibility-hidden">Edit</span>
+    <svg width="24" height="24" aria-hidden="true">
+      <use xlink:href="/icons/svg/sprite.svg#pencil" />
+    </svg>
+  </button>
+  <!-- Custom action: end -->
+  <button type="button" class="FileUploaderAttachment__action">
+    <span class="accessibility-hidden">Remove</span>
+    <svg width="24" height="24" aria-hidden="true">
+      <use xlink:href="/icons/svg/sprite.svg#close" />
+    </svg>
+  </button>
+</li>
+```
+
+#### Attaching Functionality
+
+As long as the FileUploaderAttachment lives inside the `<template>` tag, no functionality can be attached to it.
+
+👉 Don't use IDs to attach functionality to custom actions. IDs must be unique, and the FileUploaderAttachment template
+is used for every attachment.
+
+To attach functionality, you need to do so after the `queuedFile.fileUploader` event has been fired:
+
+```js
+const editHandler = (event) => {
+  alert('Edit button clicked');
+};
+
+document.addEventListener('queuedFile.fileUploader', () => {
+  document.querySelectorAll('[data-element="edit"]').forEach((element) => {
```

Comment:
```suggestion
  document.querySelectorAll('[data-spirit-element="edit"]').forEach((element) => {
```

Reply:
This is just an example, not a part of Spirit code base.

Reply:
Ok.